rawfile/utils/lock.py

```diff
@@ -0,0 +1,25 @@
+import fcntl
+from utils.rawfile import meta_file
+
+"""
+  This can be used to ensure there's only 1 operation running at a time for a given volume.
+"""
+
+
+class VolLock:
+    def __init__(self, volume_id):
+        self.path = meta_file(volume_id)
```

Comment:
What about using a separated `.lock` file

Reply:
Maybe it will not interfere with other ops on the same file, also it will be easier to clear lock file manually